In [1]:
#importing necessary libraries
import pandas as pd
import numpy as np
from sklearn import preprocessing
from sklearn import tree
from sklearn.ensemble import RandomForestClassifier

#loading working file in variable data
data=pd.read_excel("C:/Users/Dell/Downloads/Bank_Personal_Loan_Modelling.xlsx",sheet_name=1)

In [2]:
#Reading data features
data.columns


Index(['ID', 'Age', 'Experience', 'Income', 'ZIP Code', 'Family', 'CCAvg',
       'Education', 'Mortgage', 'Personal Loan', 'Securities Account',
       'CD Account', 'Online', 'CreditCard'],
      dtype='object')

In [3]:
#Cleaning of data
data.isna().sum()

ID                    0
Age                   0
Experience            0
Income                0
ZIP Code              0
Family                0
CCAvg                 0
Education             0
Mortgage              0
Personal Loan         0
Securities Account    0
CD Account            0
Online                0
CreditCard            0
dtype: int64

In [5]:
#Declaring Features of model
rf_model=RandomForestClassifier(n_estimators=1000,max_features=2,oob_score=True)


In [6]:
#Inserting all independent variables in one variable
features=['ID', 'Age', 'Experience', 'Income', 'ZIP Code', 'Family', 'CCAvg',
       'Education', 'Mortgage', 'Securities Account',
       'CD Account', 'Online', 'CreditCard']

In [7]:
#Bulding Random forest model
rf_model.fit(X=data[features],y=data["Personal Loan"])

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features=2,
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=1000,
                       n_jobs=None, oob_score=True, random_state=None,
                       verbose=0, warm_start=False)

In [8]:
#Exporting important features from Random forest model
for feature,imp in zip(features,rf_model.feature_importances_):
    print(feature,imp);

ID 0.043555229699206124
Age 0.039070303678001184
Experience 0.03819988758598691
Income 0.3218021762318392
ZIP Code 0.040739167694830325
Family 0.08348486794769364
CCAvg 0.17233855362421024
Education 0.145377928195117
Mortgage 0.041776337238018144
Securities Account 0.005146756683995985
CD Account 0.05154895320637577
Online 0.007821779798053758
CreditCard 0.009138058416671676


In [9]:
tree_model=tree.DecisionTreeClassifier()
#Inserting all independent variables in one variable(here we took variables which has value greater than 0.5)
X_var=pd.DataFrame([data["Income"],data["CCAvg"],data["Education"]]).T
#Bulding Decision tree model
tree_model.fit(X=X_var,y=data["Personal Loan"])

DecisionTreeClassifier(ccp_alpha=0.0, class_weight=None, criterion='gini',
                       max_depth=None, max_features=None, max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, presort='deprecated',
                       random_state=None, splitter='best')

In [11]:
#finding Accuracy of model
tree_model.score(X=X_var,y=data["Personal Loan"])

0.9972

__Hence,By finding important features with help of random forest and taking it as independent variables we got the accuracy of tree model 99.72%__